src/category.rs

```diff
@@ -1,57 +1,227 @@
-use quick_xml::{XmlReader, XmlWriter, Element, Event};
-use quick_xml::error::Error as XmlError;
+// This file is part of rss.
+//
+// Copyright © 2015-2017 The rust-syndication Developers
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the MIT License and/or Apache 2.0 License.
 
+use error::Error;
 use fromxml::FromXml;
+use quick_xml::{Element, Event, XmlReader, XmlWriter};
+use quick_xml::error::Error as XmlError;
+use reqwest::Url;
 use toxml::ToXml;
-use error::Error;
 
 /// A representation of the `<category>` element.
 #[derive(Debug, Default, Clone, PartialEq)]
 pub struct Category {
     /// The name of the category.
-    pub name: String,
+    name: String,
     /// The domain for the category.
-    pub domain: Option<String>,
+    domain: Option<String>,
+}
+
+impl Category {
+    /// Get the category that exists under `Category`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::{CategoryBuilder, Category};
+    ///
+    /// let category = "podcast";
+    ///
+    /// let category_obj = CategoryBuilder::new()
+    ///     .name(category)
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// assert_eq!(category, category_obj.name());
+    /// ```
+    pub fn name(&self) -> &str {
+        self.name
+            .as_str()
+    }
+
+    /// Get the optional domain that exists under `Category`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::{CategoryBuilder, Category};
+    ///
+    /// let domain_string = "http://jupiterbroadcasting.com/".to_owned();
+    ///
+    /// let category = CategoryBuilder::new()
+    ///     .domain(Some(domain_string.clone()))
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// let domain_option = category.domain();
+    /// assert_eq!(Some(domain_string.clone()), domain_option);
+    /// ```
+    ///
+    /// ```
+    /// use rss::{CategoryBuilder, Category};
+    ///
+    /// let category = CategoryBuilder::new()
+    ///     .domain(None)
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// let domain_option = category.domain();
+    /// assert!(domain_option.is_none());
+    /// ```
+    pub fn domain(&self) -> Option<String> {
+        self.domain
```

Comment:
I think the getters that return `Option<String>` could be changed to avoid cloning as well. These could be transformed in to `Option<&str>` using something like `self.domain.as_ref().map(|s| s.as_str())`.